automation/taskcluster/lib/tasks.py

```diff
@@ -494,42 +550,45 @@ def craft_raptor_task(self, signing_task_id, mozharness_task_id, variant):
+                    ('workspace/build/blobber_upload_dir', 'test_info'),
                 )],
                 "command": [
                     "./test-linux.sh",
                     '--installer-url={}'.format(apk_location),
                     "--test-packages-url={}/{}/artifacts/public/build/target.test_packages.json".format(_DEFAULT_TASK_URL, mozharness_task_id),
-                    "--test=raptor-speedometer",
+                    "--test={}".format(test_name),
                     "--app=refbrow",
                     "--binary=org.mozilla.reference.browser",
                     "--activity=GeckoViewActivity",
                     "--download-symbols=ondemand"
-                ],
+                ] + extra_test_args,
                 "env": {
-                    "XPCOM_DEBUG_BREAK": "warn",
-                    "MOZ_NO_REMOTE": "1",
+                    "GECKO_HEAD_REPOSITORY": "https://hg.mozilla.org/mozilla-central",
+                    "GECKO_HEAD_REV": gecko_revision,
+                    "MOZ_AUTOMATION": "1",
                     "MOZ_HIDE_RESULTS_TABLE": "1",
-                    "TASKCLUSTER_WORKER_TYPE": 'proj-autophone/{}'.format(worker_type),
-                    "MOZHARNESS_URL": "{}/{}/artifacts/public/build/mozharness.zip".format(_DEFAULT_TASK_URL, mozharness_task_id),
+                    "MOZ_NO_REMOTE": "1",
+                    "MOZ_NODE_PATH": "/usr/local/bin/node",
+                    "MOZHARNESS_CONFIG": "raptor/android_hw_config.py",
                     "MOZHARNESS_SCRIPT": "raptor_script.py",
+                    "MOZHARNESS_URL": "{}/{}/artifacts/public/build/mozharness.zip".format(_DEFAULT_TASK_URL, mozharness_task_id),
+                    "MOZILLA_BUILD_URL": apk_location,
                     "NEED_XVFB": "false",
+                    "NO_FAIL_ON_TEST_ERRORS": "1",
+                    "TASKCLUSTER_WORKER_TYPE": 'proj-autophone/{}'.format(worker_type),
                     "WORKING_DIR": "/builds/worker",
                     "WORKSPACE": "/builds/worker/workspace",
-                    "MOZ_NODE_PATH": "/usr/local/bin/node",
-                    "NO_FAIL_ON_TEST_ERRORS": "1",
-                    "MOZHARNESS_CONFIG": "raptor/android_hw_config.py",
-                    "MOZ_AUTOMATION": "1",
-                    "MOZILLA_BUILD_URL": apk_location,
+                    "XPCOM_DEBUG_BREAK": "warn",
                 },
                 "context": "https://hg.mozilla.org/mozilla-central/raw-file/{}/taskcluster/scripts/tester/test-linux.sh".format(gecko_revision)
             },
             treeherder={
                 'jobKind': 'test',
+                'groupSymbol': 'Rap' if group_symbol is None else group_symbol,
                 'machine': {
-                  'platform': _craft_treeherder_platform_from_variant(variant),
+                  'platform': 'android-hw-g5-7-0-arm7-api-16' if architecture == 'arm' else 'android-hw-p2-8-0-arm7-api-16',
```

Comment:
so technically we have:
arm7 on:
* android-hw-g5-7-0-arm7-api-16
* android-hw-p2-8-0-arm7-api-16

and aarch64:
* android-hw-p2-8-0-aarch64

will it be possible to support all 3?  If we change our platform, is this discoverable?  I would prefer a constants file or something in a common section of hardcoded things (platform names, job names, treeherder symbols, env vars, params) - this is really close, but I do get concerned with maintenance.

Reply:
Thanks for reminding me the "arm on pixel2" scenario. I just added it.